### PR TITLE
Upload video

### PR DESCRIPTION
**Pull request by weaselix**

### CODE REVIEW

File: TranscodingDemo/Server/uploader/utils.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    @classmethod
+    def from_request_data(cls, data):
+        return cls(int(data[FineUploaderParams.PART.value]), int(data[FineUploaderParams.CHUNK_SIZE.value]),
+                   int(data[FineUploaderParams.OFFSET.value]))
+
+
+def exception_to_fine_client_response(exception):
+    try:
+        raise exception
```
Comment (bidzyyys) [inline TranscodingDemo/Server/uploader/utils.py]: I do not like this rearise. I would write one method per one exception type. It is possible with https://pypi.org/project/multipledispatch/

File: TranscodingDemo/Server/uploader/utils.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        return FailResponse(status_code=http.HTTPStatus.BAD_REQUEST, error_msg=e.message)
+    except TooManyRequests:
+        return FailResponse(status_code=http.HTTPStatus.TOO_MANY_REQUESTS,
+                            error_msg='Maximal number of trials was exceeded', prevent_retry=True)
+    except (Exception, OperationalError, IntegrityError):
+        return FailResponse(status_code=http.HTTPStatus.INTERNAL_SERVER_ERROR,
+                            error_msg='An internal error has occurred', prevent_retry=True)
+
+
+def reset_or_too_many_requests(resource, error_msg):
```
Comment (bidzyyys) [inline TranscodingDemo/Server/uploader/utils.py]: One method, one functionality. IMO should be divided into two independent  methods.

File: TranscodingDemo/Server/uploader/views.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+                    self._clear_chunk_storage(resource, str(job.id))
+                chunk_file = request.FILES[FineUploaderParams.FILE.value]
+                self.check_and_save_chunk(chunk_file, chunk_request, resource)
+            return SuccessResponse()
+
+        except ResponseException as e:
+            logger.error(e.response_message)
+            return e.to_response()
+        except Exception as e:
+            logger.exception('An error occurred')
```
Comment (bidzyyys) [inline TranscodingDemo/Server/uploader/views.py]: Catching exception `as e` is redundant here, you do not use it.

File: TranscodingDemo/Server/uploader/views.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        if not job_id:
+            job = TranscodingJob.objects.create(resource=Resource.from_dict(request.POST))
+            file = self.storage.get_chunk_storage(str(job.id))
+            job.resource.path = file.name
+            request.session['transcoding_job'] = str(job.id)
+            return job
+        return TranscodingJob.objects.get(id=job_id)
+
+    def check_and_save_chunk(self, chunk_file, chunk_request, resource):
+        if chunk_request.part > resource.total_parts or chunk_request.offset > resource.total_size:
```
Comment (bidzyyys) [inline TranscodingDemo/Server/uploader/views.py]: One method, one functionality. I would write one method for validation and one for saving.

File: TranscodingDemo/Server/uploader/utils.py (hunk 1 above)
Comment (weaselix) [reply TranscodingDemo/Server/uploader/utils.py]: Yes, I dislike it too. That method was just something like a template (this method is not used - I forgot to remove that)

File: TranscodingDemo/Server/uploader/utils.py (hunk 2 above)
Comment (weaselix) [reply TranscodingDemo/Server/uploader/utils.py]: That method is responsible for one functionality actually: it returns either ResetUploadException  or TooManyRequests.

**Pull request merged**